mopidy/core/tracklist.py

```diff
@@ -646,3 +647,33 @@ def _trigger_tracklist_changed(self):
     def _trigger_options_changed(self):
         logger.debug('Triggering options changed event')
         listener.CoreListener.send('options_changed')
+
+    def _export_state(self):
+        """Internal method for :class:`mopidy.Core`."""
+        return TracklistState(
+            tl_tracks=self._tl_tracks,
+            next_tlid=self._next_tlid,
+            consume=self.get_consume(),
+            random=self.get_random(),
+            repeat=self.get_repeat(),
+            single=self.get_single())
+
+    def _restore_state(self, state, coverage):
+        """Internal method for :class:`mopidy.Core`."""
+        if state:
+            if 'mode' in coverage:
+                self.set_consume(state.consume)
+                self.set_random(state.random)
+                self.set_repeat(state.repeat)
+                self.set_single(state.single)
+            if 'tracklist' in coverage:
+                if state.next_tlid > self._next_tlid:
+                    self._next_tlid = state.next_tlid
+                self._tl_tracks = []
+                for track in state.tl_tracks:
+                    # TODO: check if any backend will play the track.
+                    # Could be an issue with music streaming services
+                    # (login), disabled extensions and automatically
+                    # generated playlists (pandora).
```

Comment:
I don't think we should care about the play-ability of the track when restoring state. It would take a bunch of time to test properly, all backends are probably not up and running fully at the early stage we restore state. Unplayable tracks are handled when we try to play them. I think the TODO can be removed.